nibabel/streamlines/tck.py

```diff
@@ -0,0 +1,464 @@
+    convention as MRtrix.
+
+    References
+    ----------
+    [1] http://www.nitrc.org/pipermail/mrtrix-discussion/2014-January/000859.html
+    [2] http://nipy.org/nibabel/coordinate_systems.html#voxel-coordinates-are-in-voxel-space
+    """
+    # Constants
+    MAGIC_NUMBER = "mrtrix tracks"
+    SUPPORTS_DATA_PER_POINT = False  # Not yet
+    SUPPORTS_DATA_PER_STREAMLINE = False  # Not yet
+
+    FIBER_DELIMITER = np.array([[np.nan, np.nan, np.nan]], '<f4')
+    EOF_DELIMITER = np.array([[np.inf, np.inf, np.inf]], '<f4')
+
+    def __init__(self, tractogram, header=None):
+        """
+        Parameters
+        ----------
+        tractogram : :class:`Tractogram` object
+            Tractogram that will be contained in this :class:`TckFile`.
+        header : None or dict, optional
+            Metadata associated to this tractogram file. If None, make
+            default empty header.
+
+        Notes
+        -----
+        Streamlines of the tractogram are assumed to be in *RAS+* and *mm*
+        space. It is also assumed that when streamlines are mapped back to
+        voxel space, a streamline point located at an integer coordinate
+        (i,j,k) is considered to be at the center of the corresponding voxel.
+        This is in contrast with TRK's internal convention where it would
+        have referred to a corner.
+        """
+        if header is None:
+            header = self.create_empty_header()
+
+        super(TckFile, self).__init__(tractogram, header)
+
+    @classmethod
+    def is_correct_format(cls, fileobj):
+        """ Check if the file is in TCK format.
+
+        Parameters
+        ----------
+        fileobj : string or file-like object
+            If string, a filename; otherwise an open file-like object
+            pointing to TCK file (and ready to read from the beginning
+            of the TCK header data). Note that calling this function
+            does not change the file position.
+
+        Returns
+        -------
+        is_correct_format : {True, False}
+            Returns True if `fileobj` is compatible with TCK format,
+            otherwise returns False.
+        """
+        with Opener(fileobj) as f:
+            magic_number = asstr(f.fobj.readline())
+            f.seek(-len(magic_number), os.SEEK_CUR)
+
+        return magic_number.strip() == cls.MAGIC_NUMBER
+
+    @classmethod
+    def create_empty_header(cls):
+        """ Return an empty compliant TCK header. """
+        header = {}
+
+        # Default values
+        header[Field.MAGIC_NUMBER] = cls.MAGIC_NUMBER
+        header[Field.NB_STREAMLINES] = 0
+        header['datatype'] = "Float32LE"
+        return header
+
+    @classmethod
+    def load(cls, fileobj, lazy_load=False):
+        """ Loads streamlines from a filename or file-like object.
+
+        Parameters
+        ----------
+        fileobj : string or file-like object
+            If string, a filename; otherwise an open file-like object
```

Comment:
I don't know what happened, but you are right,t it was missing some docstrings that weren't specifying binary mode. Should be fixed by commit 0d0ece18b83b0e03079ca6bedac9bef6d5c3a3e8.